### PR TITLE
Implement 'getPortfolioStep' function

### PR DESCRIPTION
**Pull request by zachclark-ccpo**
Ticket: **AT-6363**

### CODE REVIEW

File: infrastructure/lib/atat-web-api-stack.ts [hunk 1]
```diff
@@ -113,9 +113,17 @@ export class AtatWebApiStack extends cdk.Stack {
...
+      entry: "applications/portfolioDrafts/portfolio/getPortfolioStep.ts",
+      ...sharedFunctionProps,
+    });
+    portfolio.addMethod("GET", new apigw.LambdaIntegration(getPortfolioStepFn));
+    table.grantReadData(getPortfolioStepFn);
+
     // TODO: getPortfolioDrafts
-    // TODO: getPortfolioDraft
     // TODO: getPortfolioStep
+    // TODO: getPortfolioDraft
```
Comment (kylelaker-ccpo) [inline infrastructure/lib/atat-web-api-stack.ts:126]: I think you meant to delete getPortfolioStep?

Comment (zachclark-ccpo) [reply infrastructure/lib/atat-web-api-stack.ts:126]: Yes

File: applications/portfolioDrafts/portfolio/getPortfolioStep.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import { PortfolioStep } from "../models/PortfolioStep";
+import { dynamodbDocumentClient as client } from "../utils/dynamodb";
+import { ApiSuccessResponse, ErrorResponse, ErrorStatusCode, SuccessStatusCode } from "../utils/response";
+
+const TABLE_NAME = process.env.ATAT_TABLE_NAME;
+const NO_SUCH_PORTFOLIO_STEP = new ErrorResponse(
+  { code: ErrorCodes.INVALID_INPUT, message: "Portfolio Step not found for this Portfolio Draft" },
+  ErrorStatusCode.NOT_FOUND
+);
+const NO_SUCH_PORTFOLIO = new ErrorResponse(
```
Comment (jasonburkert-ccpo) [inline applications/portfolioDrafts/portfolio/getPortfolioStep.ts:13]: I want to be careful about the Portfolio/PortfolioDraft language.  Can we call this...
```suggestion
const NO_SUCH_PORTFOLIO_DRAFT = new ErrorResponse(
```

**Review by kylelaker-ccpo — APPROVED** [review]

**Review by jasonburkert-ccpo — APPROVED** [review]
Looks good.  Tried deployed instance.

**Comment by jasonburkert-ccpo** [comment]
12 seconds late.

**Pull request merged**